src/gui/attributetable/qgsfeaturelistview.cpp

```diff
@@ -183,6 +183,17 @@ void QgsFeatureListView::editSelectionChanged( const QItemSelection &deselected,
     viewport()->update( visualRegionForSelection( localDeselected ) | visualRegionForSelection( localSelected ) );
   }
 
+  if ( !selected.isEmpty() )
+  {
+    const QModelIndexList indexList = selected.indexes();
+    if ( !indexList.isEmpty() )
+    {
+      QgsFeature selectedFeature;
+      mModel->featureByIndex( mModel->mapFromMaster( indexList.first() ), selectedFeature );
+      mLastEditSelectionFid = selectedFeature.id();
+    }
+  }
```

Comment:
Else set to null fid?

Reply:
shouldn't happen. But yes to QgsFeatureId()...

Reply:
if it _really_ shouldn't happen you can also add a `Q_ASSERT( false )`

Reply:
I'm not 100% sure. In the workflows I know it does not happen. But it's the attribute table. When it's null just nothing happens (index might jump to 0 like before...) so I would feel better with it...